src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -381,6 +391,23 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return x;
 	}
 
+	/** Eigen3 QR solver */
+	template <typename T>
+	SGMatrix<T> qr_solver_impl(const SGMatrix<T>& A, const SGMatrix<T>& b) const
+	{
+		SGMatrix<T> x = SGMatrix<T>(b.num_rows, b.num_cols);
+		x.zero();
+
+		typename SGMatrix<T>::EigenMatrixXtMap A_eig = A;
+		typename SGMatrix<T>::EigenMatrixXtMap b_eig = b;
+		typename SGMatrix<T>::EigenMatrixXtMap x_eig = x;
+
+		Eigen::HouseholderQR<Eigen::Matrix<T,Eigen::Dynamic,Eigen::Dynamic> > qr(A_eig);
```

Comment:
We want to offer the other variants in eigen3 as well. At least the more stable ones with pivoting.
We can do that via an enum for each solver type.....

See e.g. https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.linalg.qr.html
They also offer different methods